randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -52,16 +61,56 @@ def force_field_index_for_requirement(game: GameDescription, requirement: Layout
     "Octolith8": 7,
 }
 
+_ITEM_TO_ITEM_TYPE = {
+    "Energy Tank": 4,
+    "Volt Driver": 5,
+    "Missile": 6,
+    "Battlehammer": 7,
+    "Imperialist": 8,
+    "Judicator": 9,
+    "Magmaul": 10,
+    "Shock Coil": 11,
+}
+
 
 class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosmeticPatches]):
     def game_enum(self) -> RandovaniaGame:
         return RandovaniaGame.METROID_PRIME_HUNTERS
 
-    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
-        result = {
-            # TODO: Take starting weapons into account
-            "weapons_string": "00000101",
-        }
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str | int]:
+        result: dict[str, str | int] = {}
+        starting_items = {}
+        resource_gain = [(resource, quantity) for resource, quantity in resources.as_resource_gain()]
+
+        for resource, quantity in resource_gain:
+            try:
+                starting_items[item_type_for_item_resource(resource)] = quantity
+            except KeyError:
+                print(f"Skipping {resource} for starting inventory: no item id")
+                continue
+
+        weapons = str(
+            [
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Shock Coil"], 0),
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Magmaul"], 0),
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Judicator"], 0),
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Imperialist"], 0),
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Battlehammer"], 0),
+                1,  # Missiles
+                starting_items.get(_ITEM_TO_ITEM_TYPE["Volt Driver"], 0),
+                1,  # Power Beam
+            ]
+        )
+        starting_weapons = "".join(filter(str.isdigit, weapons))
```

Comment:
what's the point of the filter here?

Reply:
I need to pass a bitfield as a string to the patcher.

Reply:
I don't understand. Why does this require a filter?